Interop Predeploys: implement L2 Genesis

0xfuturistic opened the pull request.

0xfuturistic committed:
op-chain-ops: add CrossL2Inbox to genesis config

0xfuturistic committed:
op-chain-ops: add CrossL2Inbox to immutables

0xfuturistic committed:
op-chain-ops: add CrossL2Inbox to layer_two_test

0xfuturistic committed:
op-chain-ops: add CrossL2Inbox to immutables_test

0xfuturistic committed:
contracts-bedrock: add CrossL2Inbox to L2Genesis

0xfuturistic committed:
op-chain-ops: add L2ToL2CrossDomainMessenger to genesis config

0xfuturistic committed:
op-chain-ops: add L2ToL2CrossDomainMessenger to immutables

0xfuturistic committed:
op-chain-ops: update value in layer_two_test

0xfuturistic committed:
op-chain-ops: add L2ToL2CrossDomainMessenger to immutables_test

0xfuturistic committed:
contracts-bedrock: add L2ToL2CrossDomainMessenger to L2Genesis

0xfuturistic committed:
contracts-bedrock: fix missing bracket in L2Genesis.s.sol

tynes commented:
This looks good to me, we can rebase this on top of the other PRs when they are merged

0xfuturistic committed:
op-chain-ops: fix layer_two_test.go

roninjin10 committed:
fix: make sdk-next tests run in ci (#10003)

Co-authored-by: Will Cory <willcory@Wills-MacBook-Pro.local>

roninjin10 committed:
chore: Release new versions of all js packages (#9964)

Co-authored-by: Will Cory <willcory@Wills-MacBook-Pro.local>

dependabot[bot] committed:
build(deps): bump google.golang.org/protobuf in /op-ufm (#9854)

Bumps google.golang.org/protobuf from 1.32.0 to 1.33.0.

---
updated-dependencies:
- dependency-name: google.golang.org/protobuf
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

keroro520 committed:
indexer: update format string for uuid.UUID (#9772)

* indexer: update format string for uuid.UUID

The relayEvent variable is of type uuid.UUID, but it's being formatted as an
integer (%d). This will cause a runtime error. Use %s to format the UUID as a
string.

* indexer: constraint l2_block_headers.timestamp > 0

* indexer: doc bigint.Clamp assumption

refcell committed:
feat(op-dispute-mon): GetWithdrawals GameCaller Method (#9939)

* feat(op-challenger): Delayed Weth Withdrawal Request Caller

* fix(op-challenger): withdrawal request field ordering

* fix(op-challenger): encapsulate delayed weth behind the fault dispute game contract binding

* feat(op-dispute-mon): get withdrawals game caller method

* fix(op-dispute-mon): revert service name changes

* fix(op-dispute-mon): revert extractor test name

* fix(op-dispute-mon): revert extractor changes

* fix(op-dispute-mon): revert newline add

* fix(op-dispute-mon): revert caller field reordering

* fix(op-challenger): make contract method private

mds1 committed:
test(ctb): mark coverage as successful if it fails with stack too deep (#10008)

* test(ctb): mark coverage as successful if it fails with stack too deep

* Update packages/contracts-bedrock/package.json

refcell committed:
feat(op-dispute-mon): WithdrawalRequest and Recipients Enricher (#9940)

* feat(op-challenger): Delayed Weth Withdrawal Request Caller

* fix(op-challenger): withdrawal request field ordering

* fix(op-challenger): encapsulate delayed weth behind the fault dispute game contract binding

* feat(op-dispute-mon): get withdrawals game caller method

* fix(op-dispute-mon): revert service name changes

* fix(op-dispute-mon): revert extractor test name

* fix(op-dispute-mon): revert extractor changes

* fix(op-dispute-mon): revert newline add

* fix(op-dispute-mon): revert caller field reordering

* feat(op-dispute-mon): weth caller creation and pass it through sub components

* chore(op-dispute-mon): weth caller creation unit test

* fix(op-dispute-mon): extractor test for weth caller creation error

* feat(op-dispute-mon): wire up the withdrawals extractor

* fix(op-dispute-mon): rebases

* fix(op-dispute-mon): revert caller test change

* fix(op-dispute-mon): recipient enricher

refcell committed:
feat(op-dispute-mon): WithdrawalRequests Metrics (#9942)

* feat(op-challenger): Delayed Weth Withdrawal Request Caller

* fix(op-challenger): withdrawal request field ordering

* fix(op-challenger): encapsulate delayed weth behind the fault dispute game contract binding

* feat(op-dispute-mon): get withdrawals game caller method

* fix(op-dispute-mon): revert service name changes

* fix(op-dispute-mon): revert extractor test name

* fix(op-dispute-mon): revert extractor changes

* fix(op-dispute-mon): revert newline add

* fix(op-dispute-mon): revert caller field reordering

* feat(op-dispute-mon): weth caller creation and pass it through sub components

* chore(op-dispute-mon): weth caller creation unit test

* fix(op-dispute-mon): extractor test for weth caller creation error

* feat(op-dispute-mon): wire up the withdrawals extractor

* fix(op-dispute-mon): rebases

* fix(op-dispute-mon): revert caller test change

* fix(op-dispute-mon): recipient enricher

* feat(op-dispute-mon): withdrawal request metrics

* Update op-dispute-mon/mon/withdrawals.go

Co-authored-by: Adrian Sutton <adrian@oplabs.co>

* fix(op-dispute-mon): withdrawal request amounts

---------

Co-authored-by: Adrian Sutton <adrian@oplabs.co>

Commit:
all: use the built-in slices library (#10005)

Signed-off-by: carehabit <shenyuting@outlook.com>

0xfuturistic commented:
new PR: https://github.com/ethereum-optimism/optimism/pull/10294
